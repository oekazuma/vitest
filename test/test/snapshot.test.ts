import { expect, test } from 'vitest'

test('snapshot', () => {
  expect({
    this: { is: new Set(['of', 'snapshot']) },
  }).toMatchSnapshot()
})

test('inline snapshot', () => {
  expect('inline string').toMatchInlineSnapshot('"inline string"')
  expect({ foo: { type: 'object', map: new Map() } }).toMatchInlineSnapshot(`
{
  "foo": {
    "map": Map {},
    "type": "object",
  },
}`)
})

test('snapshot with big array', () => {
  expect({
    this: { is: new Set(['one', new Array(30).fill({})]) },
  }).toMatchSnapshot()
})

test('snapshot with big string', () => {
  expect({
    this: { is: new Set(['one', new Array(30).fill('zoo').join()]) },
  }).toMatchSnapshot()
})

test('throwing snapshots', () => {
  expect(() => {
    throw new Error('omega')
  }).toThrowErrorMatchingSnapshot()

  expect(() => {
    // eslint-disable-next-line no-throw-literal
    throw 'omega'
  }).toThrowErrorMatchingSnapshot()

  expect(() => {
    // eslint-disable-next-line no-throw-literal
    throw { error: 'omega' }
  }).toThrowErrorMatchingSnapshot()
})

test('throwing inline snapshots', () => {
  expect(() => {
    throw new Error('omega')
  }).toThrowErrorMatchingInlineSnapshot('"omega"')

  expect(() => {
    // eslint-disable-next-line no-throw-literal
    throw 'omega'
  }).toThrowErrorMatchingInlineSnapshot('"omega"')

  expect(() => {
    // eslint-disable-next-line no-throw-literal
    throw { error: 'omega' }
  }).toThrowErrorMatchingInlineSnapshot(`
{
  "error": "omega",
}
`)
})
