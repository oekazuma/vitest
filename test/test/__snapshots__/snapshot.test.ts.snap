// Vitest Snapshot v1

exports[`snapshot 1`] = `
{
  "this": {
    "is": Set {
      "of",
      "snapshot",
    },
  },
}
`;

exports[`snapshot with big array 1`] = `
{
  "this": {
    "is": Set {
      "one",
      [
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
      ],
    },
  },
}
`;

exports[`snapshot with big string 1`] = `
{
  "this": {
    "is": Set {
      "one",
      "zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo",
    },
  },
}
`;

exports[`throwing snapshots 1`] = `"omega"`;

exports[`throwing snapshots 2`] = `"omega"`;

exports[`throwing snapshots 3`] = `
{
  "error": "omega",
}
`;
